File: android/app/src/main/java/com/reservationapp/ui/admin/reservations/ReservationRowView.kt
package com.reservationapp.ui.admin.reservations

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CalendarToday
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.reservationapp.core.ui.theme.*
import com.reservationapp.domain.model.Reservation
import com.reservationapp.ui.admin.components.StatusBadge
import java.text.SimpleDateFormat
import java.util.*

@Composable
fun ReservationRowView(
    reservation: Reservation,
    onDetailTap: () -> Unit,
    onApproveTap: (() -> Unit)?,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = BackgroundPaper
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(Spacing.md),
            verticalArrangement = Arrangement.spacedBy(Spacing.md)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.Top
            ) {
                Column(
                    modifier = Modifier.weight(1f),
                    verticalArrangement = Arrangement.spacedBy(Spacing.xs)
                ) {
                    // Reservation Number
                    Text(
                        text = reservation.reservationNumber,
                        style = MaterialTheme.typography.bodyLarge.copy(
                            fontWeight = FontWeight.Medium
                        ),
                        color = TextPrimary
                    )

                    // User Name
                    Text(
                        text = reservation.userName,
                        style = MaterialTheme.typography.bodyMedium,
                        color = TextSecondary
                    )

                    // Date and Time Slot
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(Spacing.sm),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(
                            imageVector = Icons.Default.CalendarToday,
                            contentDescription = null,
                            modifier = Modifier.size(14.dp),
                            tint = TextSecondary
                        )
                        Text(
                            text = formatDate(reservation.date),
                            style = MaterialTheme.typography.labelSmall,
                            color = TextSecondary
                        )

                        Spacer(modifier = Modifier.width(Spacing.sm))

                        Icon(
                            imageVector = Icons.Default.Schedule,
                            contentDescription = null,
                            modifier = Modifier.size(14.dp),
                            tint = TextSecondary
                        )
                        Text(
                            text = getTurkishMealTimeSlotName(reservation.mealTimeSlotName),
                            style = MaterialTheme.typography.labelSmall,
                            color = TextSecondary
                        )
                    }

                    // Restaurant and Menu
                    Text(
                        text = "${reservation.restaurantName} • ${reservation.menuName.takeIf { it.isNotEmpty() } ?: "Menü"}",
                        style = MaterialTheme.typography.labelSmall,
                        color = TextTertiary
                    )
                }

                Column(
                    horizontalAlignment = Alignment.End,
                    verticalArrangement = Arrangement.spacedBy(Spacing.sm)
                ) {
                    // Status Badge
                    StatusBadge(status = reservation.status)

                    // Actions
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(Spacing.sm)
                    ) {
                        TextButton(onClick = onDetailTap) {
                            Text(
                                text = "Detay",
                                style = MaterialTheme.typography.labelSmall,
                                color = PrimaryMain
                            )
                        }

                        if (isPending(reservation.status) && onApproveTap != null) {
                            Button(
                                onClick = onApproveTap,
                                colors = ButtonDefaults.buttonColors(
                                    containerColor = InfoMain
                                ),
                                modifier = Modifier.height(28.dp),
                                contentPadding = PaddingValues(horizontal = Spacing.sm, vertical = Spacing.xs)
                            ) {
                                Text(
                                    text = "Onayla",
                                    style = MaterialTheme.typography.labelSmall
                                )
                            }
                        }
                    }
                }
            }
        }
    }
}

private fun formatDate(dateString: String): String {
    return try {
        val inputFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.getDefault())
        val outputFormat = SimpleDateFormat("d MMMM yyyy", Locale("tr", "TR"))
        val date = inputFormat.parse(dateString) ?: return dateString
        outputFormat.format(date)
    } catch (e: Exception) {
        dateString
    }
}

private fun getTurkishMealTimeSlotName(englishName: String): String {
    return when (englishName.lowercase()) {
        "breakfast" -> "Kahvaltı"
        "lunch" -> "Öğle Yemeği"
        "dinner" -> "Akşam Yemeği"
        else -> englishName
    }
}

private fun isPending(status: String): Boolean {
    return status.lowercase() == "pending" || status == "3"
}

File: ios/ReservationApp/ReservationApp/Presentation/Views/Admin/Reservations/Components/ReservationRowView.swift
//
//  ReservationRowView.swift
//  ReservationApp
//
//  Created by Mehmet Ali Kısacık on 13.11.2025.
//

import SwiftUI

struct ReservationRowView: View {
    let reservation: Reservation
    let onDetailTap: () -> Void
    let onApproveTap: (() -> Void)?
    
    var body: some View {
        VStack(spacing: ThemeManager.shared.spacing.md) {
            HStack(alignment: .top) {
                VStack(alignment: .leading, spacing: ThemeManager.shared.spacing.xs) {
                    // Reservation Number
                    Text(reservation.reservationNumber)
                        .font(AppTypography.body1())
                        .fontWeight(.medium)
                        .foregroundColor(AppColors.textPrimary)
                    
                    // User Name
                    Text(reservation.userName)
                        .font(AppTypography.body2())
                        .foregroundColor(AppColors.textSecondary)
                    
                    // Date and Time Slot
                    HStack(spacing: ThemeManager.shared.spacing.sm) {
                        Label(formatDate(reservation.date), systemImage: "calendar")
                            .font(AppTypography.caption())
                            .foregroundColor(AppColors.textSecondary)
                        
                        Label(reservation.mealTimeSlotName, systemImage: "clock")
                            .font(AppTypography.caption())
                            .foregroundColor(AppColors.textSecondary)
                    }
                    
                    // Restaurant and Menu
                    Text("\(reservation.restaurantName) • \(reservation.menuName.isEmpty ? "Menü" : reservation.menuName)")
                        .font(AppTypography.caption())
                        .foregroundColor(AppColors.textTertiary)
                }
                
                Spacer()
                
                VStack(alignment: .trailing, spacing: ThemeManager.shared.spacing.sm) {
                    // Status Badge
                    if let status = ReservationStatus(rawValue: reservation.status) {
                        StatusBadge(status: status)
                    } else {
                        // Handle numeric status or unknown status
                        StatusBadge(status: .pending)
                    }
                    
                    // Actions
                    HStack(spacing: ThemeManager.shared.spacing.sm) {
                        Button(action: onDetailTap) {
                            Text("Detay")
                                .font(AppTypography.caption())
                                .foregroundColor(AppColors.primaryMain)
                        }
                        
                        if isPending(reservation.status) {
                            if let onApprove = onApproveTap {
                                Button(action: onApprove) {
                                    Text("Onayla")
                                        .font(AppTypography.caption())
                                        .foregroundColor(.white)
                                        .padding(.horizontal, ThemeManager.shared.spacing.sm)
                                        .padding(.vertical, ThemeManager.shared.spacing.xs)
                                        .background(AppColors.infoMain)
                                        .cornerRadius(ThemeManager.shared.borderRadius.button.value)
                                }
                            }
                        }
                    }
                }
            }
        }
        .padding(ThemeManager.shared.spacing.md)
        .background(AppColors.backgroundPaper)
        .cornerRadius(ThemeManager.shared.borderRadius.card.value)
        .shadow(color: Color.black.opacity(0.05), radius: 4, x: 0, y: 2)
    }
    
    private func formatDate(_ dateString: String) -> String {
        // Format as "4 Kasım 2025"
        if let date = ISO8601DateFormatter().date(from: dateString) {
            let formatter = DateFormatter()
            formatter.locale = Locale(identifier: "tr_TR")
            formatter.dateFormat = "d MMMM yyyy"
            return formatter.string(from: date)
        }
        return dateString // Fallback
    }
    
    private func isPending(_ status: String) -> Bool {
        return status.lowercased() == "pending" || status == "3"
    }
}



